src/Nix/Derivation/Builder.hs

```diff
@@ -6,29 +6,33 @@
 module Nix.Derivation.Builder
     ( -- * Builder
       buildDerivation
+    , buildDerivation'
     ) where
 
-import Data.Foldable (foldMap)
 import Data.Map (Map)
-import Data.Monoid ((<>))
 import Data.Set (Set)
 import Data.Text (Text)
 import Data.Text.Lazy.Builder (Builder)
 import Data.Vector (Vector)
-import Filesystem.Path.CurrentOS (FilePath)
 import Nix.Derivation.Types (Derivation(..), DerivationOutput(..))
-import Prelude hiding (FilePath)
 
 import qualified Data.Map
 import qualified Data.Set
 import qualified Data.Text
 import qualified Data.Text.Lazy.Builder
 import qualified Data.Vector
-import qualified Filesystem.Path.CurrentOS
 
 -- | Render a derivation as a `Builder`
-buildDerivation :: Derivation -> Builder
-buildDerivation (Derivation {..}) =
+buildDerivation :: Derivation FilePath Text -> Builder
+buildDerivation = buildDerivation' filepath' string'
+
+-- | Render a derivation as a `Builder` using custom
+-- renderer for filepath and string
+buildDerivation' :: (fp -> Builder)
```

Comment:
Maybe call this `buildDerivationWith`.  I try to avoid using `'`s in public API names

Same comment for `parseDerivation'`

Reply:
Addressed, thanks for the suggestion! I was trying to figure out a better name but only came up with `principledBuildDerivation` which is a bit too long and prefixed instead :)